### PR TITLE
[BDHK-189] Validate intents from solver before executing them onchain

### PR DESCRIPTION
**Pull request by adelowo**
Add validation step by communicating with the solver before any form of onchain execution.

### CODE REVIEW

File: solution/validate.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"net/http"
+	"net/url"
+	"unsafe"
+
+	"github.com/blndgs/model"
+	"github.com/goccy/go-json"
+	"github.com/stackup-wallet/stackup-bundler/pkg/modules"
+)
+
+func (ei *IntentsHandler) ValidateIntents() modules.BatchHandlerFunc {
```
Comment (hpsing) [inline solution/validate.go:18]: godoc

File: solution/validate.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	return func(ctx *modules.BatchHandlerCtx) error {
+
+		batchIntentIndices := make(batchIntentIndices)
+
+		modelUserOps := *(*[]*model.UserOperation)(unsafe.Pointer(&ctx.Batch))
```
Comment (hpsing) [inline solution/validate.go:23]: safer conversion if possible
```suggestion
		batchIntentIndices := make(batchIntentIndices)

		modelUserOps := *(*[]*model.UserOperation)(unsafe.Pointer(&ctx.Batch))

		modelUserOps, ok := ctx.Batch.([]*model.UserOperation)
		if !ok {
			return fmt.Errorf("failed to convert ctx.Batch to []*model.UserOperation")
		}
```

Comment (adelowo) [reply solution/validate.go:23]: this wouldn't work actually

Comment (hpsing) [reply solution/validate.go:23]: yeah correct, `ctx.Batch` is already of type`[]*userop.UserOperation`, and there is no need for type assertion. My bad.

**Review by hpsing — APPROVED** [review]

**Review by hpsing — COMMENTED** [review]
LGTM! 💯

**Pull request merged**